src/cellink/widgets/codeeditor.cpp

```diff
@@ -295,3 +297,13 @@ void LineNumberBar::updateSize(int blockCount)
     m_size.setWidth(2 * m_hzMargin + fontMetrics().horizontalAdvance(QLatin1Char('9')) * digits);
     updateGeometry();
 }
+
+void CodeEditor::updateHighlightLines()
+{
+    if (!textCursor().hasSelection()) {
+        m_selectedLines = QSet{textCursor().blockNumber()};
+    } else {
+        m_selectedLines.insert(textCursor().blockNumber());
+    }
+    emit highlightedLinesChanged(m_selectedLines);
```

Comment:
does it always change though? :)

Reply:
Good question... With `::cursorPositionChanged` it will not always change if it just moves on the same line. I will investigate.

Reply:
There was actually a bug in the implementation as well when deselecting lines